### PR TITLE
Add `AllDiag` and `AnyDiag` to helper/validation for combining `SchemaValidateDiagFunc`

### PR DESCRIPTION
**Pull request by kangasta**
Implements #780

### CODE REVIEW

File: helper/validation/testing.go [hunk 1]
```diff
@@ -15,23 +17,53 @@ type testCase struct {
 	expectedErr *regexp.Regexp
 }
 
-func runTestCases(t testing.T, cases []testCase) {
+func runTestCases(t *testing.T, cases []testCase) {
 	t.Helper()
 
 	for i, tc := range cases {
-		_, errs := tc.f(tc.val, "test_property")
+		t.Run(fmt.Sprintf("TestCase_%d", i), func(t *testing.T) {
```
Comment (kangasta) [inline helper/validation/testing.go:24]: Update also existing testcases to use `t.Run(...)`.

File: .changes/unreleased/ENHANCEMENTS-20230222-001027.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+kind: ENHANCEMENTS
+body: Add `SchemaValidateDiagFunc` variants of helper/validation `All` and `Any` as
+  `AllDiag` and `AnyDiag`.
```
Comment (bflad) [inline .changes/unreleased/ENHANCEMENTS-20230222-001027.yaml:3]:
```suggestion
body: 'helper/validation: Added `AllDiag` and `AnyDiag`, which are `SchemaValidateDiagFunc`
  variants of `All` and `Any`'
```

**Review by bflad — APPROVED** [review]
Thank you for this contribution, @kangasta, and apologies for the delayed review. Looks good to me with minor changelog entry update to match our typical formatting. 🚀

**Comment by bflad** [comment]
Looks like I cannot merge the suggestion into this PR, so will make that change post-merge.

**Pull request merged**